Framework/Algorithms/src/PolarizationCorrections/HeliumAnalyserEfficiency.cpp

```diff
@@ -96,56 +95,42 @@ void HeliumAnalyserEfficiency::init() {
+  Kernel::Unit_const_sptr unit = workspace->getAxis(0)->unit();
+  if (unit->unitID() != "Wavelength") {
+    errorList[PropertyNames::INPUT_WORKSPACE] = "All input workspaces must be in units of Wavelength.";
+    return;
+  }
+}
+
 /**
  * Tests that the inputs are all valid
  * @return A map containing the incorrect workspace
  * properties and an error message
  */
 std::map<std::string, std::string> HeliumAnalyserEfficiency::validateInputs() {
   std::map<std::string, std::string> errorList;
-  const std::string inputWorkspaceName = getProperty(PropertyNames::INPUT_WORKSPACE);
-  if (!AnalysisDataService::Instance().doesExist(inputWorkspaceName)) {
+  const WorkspaceGroup_sptr wsGroup = getProperty(PropertyNames::INPUT_WORKSPACE);
+
+  if (wsGroup == nullptr) {
+    errorList[PropertyNames::INPUT_WORKSPACE] = "The input workspace is not a group workspace";
+  } else if (wsGroup->size() != 4) {
     errorList[PropertyNames::INPUT_WORKSPACE] =
-        "The input workspace " + inputWorkspaceName + " does not exist in the ADS.";
+        "The input group workspace must have four periods corresponding to the four spin configurations.";
   } else {
-    const auto ws = AnalysisDataService::Instance().retrieve(getProperty(PropertyNames::INPUT_WORKSPACE));
-    if (!ws->isGroup()) {
-      errorList[PropertyNames::INPUT_WORKSPACE] = "The input workspace is not a group workspace";
-    } else {
-      const auto wsGroup = std::dynamic_pointer_cast<WorkspaceGroup>(ws);
-      if (wsGroup->size() != 4) {
-        errorList[PropertyNames::INPUT_WORKSPACE] =
-            "The input group workspace must have four periods corresponding to the four spin configurations.";
-      }
+    for (size_t i = 0; i < wsGroup->size(); ++i) {
+      MatrixWorkspace_sptr const stateWs = std::dynamic_pointer_cast<MatrixWorkspace>(wsGroup->getItem(i));
+      validateInputWorkspace(stateWs, errorList);
     }
   }
   return errorList;
 }
 
-/**
- * Explicitly calls validateInputs and throws runtime error in case
- * of issues in the input properties.
- */
-void HeliumAnalyserEfficiency::validateGroupInput() {
-  const auto results = validateInputs();
-  if (results.size() > 0) {
-    const auto result = results.cbegin();
-    throw std::runtime_error("Issue in " + result->first + " property: " + result->second);
-  }
-}
-
-bool HeliumAnalyserEfficiency::processGroups() {
-  validateGroupInput();
-  calculateAnalyserEfficiency();
-  return true;
-}
-
 void HeliumAnalyserEfficiency::exec() { calculateAnalyserEfficiency(); }
 
 void HeliumAnalyserEfficiency::calculateAnalyserEfficiency() {
   // First we extract the individual workspaces corresponding to each spin configuration from the group workspace
-  const auto groupWorkspace =
-      AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(getProperty(PropertyNames::INPUT_WORKSPACE));
+  WorkspaceGroup_sptr groupWorkspace = getProperty(PropertyNames::INPUT_WORKSPACE);
```

Comment:
Sorry, I didn't spot this the first time through, but this looks like the variable used to be marked as `const` to indicate that it isn't going to be changed. Please could we put the `const` back here? As we seem to be using "west const" in this file, we would define the variable as `const WorkspaceGroup_sptr groupWorkspace = `